src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -181,6 +181,19 @@ public static IServiceCollection AddApplicationInsightsTelemetry(this IServiceCo
         }
 
         /// <summary>
+        /// Adds an Application Insights Telemetry Processor into a service collection via a <see cref="ITelemetryProcessorFactory"/>.
+        /// </summary>
+        /// <typeparam name="T">Type of the telemetry processor to add.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> instance.</param>
+        /// <returns>
+        /// The <see cref="IServiceCollection"/>.
+        /// </returns>
+        public static IServiceCollection AddApplicationInsightsTelemetryProcessor<T>(this IServiceCollection services) where T : ITelemetryProcessor
```

Comment:
Usually we tend to have non generic overload for cases you don't have generic `T` (for example you are searching for types using reflection).